src/heightmap.cpp

```diff
@@ -22,6 +22,47 @@
 
 #include "safeguards.h"
 
+/**
+ * Maximum number of pixels for one dimension of a heightmap image.
+ *
+ * At some point the number of pixels in a heightmap becomes larger compared to
+ * the pixels that will be taken for the map. With this limit already 75% of the
+ * pixels would be discarded for the largest image and the largest map size.
+ */
+static const uint MAX_HEIGHTMAP_SIDE_LENGTH_IN_PIXELS = 2 * MAX_MAP_SIZE;
+
+/*
+ * Maximum size, in bytes, of the heightmap to consider loading.
+ *
+ * The theoretical maximum size of an image is about 16 exabytes for PNG and
+ * about 4 exabytes for a BMP image. On a 32 bit platform this will overflow
+ * size_t which could lead to memory corruption.
+ * Even on a 64 bit platform that is way larger than feasible; we could allow it
+ * on those platforms, but then the memory allocation that is done will almost
+ * certainly fail causing a crash to the desktop instead of a nice user error.
+ * So, artificially limit the maximum size of a hightmap image to allocate one
+ * gigabyte here.
+ */
+static const uint MAX_HEIGHTMAP_SIZE_IN_BYTES = 1 << 30;
+
+/**
+ * Check whether the loaded dimension of the heightmap is considered valid enough
+ * to attempt to load. In other words, the width and height are not beyond the
+ * #MAX_HEIGHTMAP_SIDE_LENGTH_IN_PIXELS limit and the number of bytes that would
+ * be allocated for the table does not exceed MAX_HEIGHTMAP_SIZE_IN_BYTES.
```

Comment:
`table`? What table?

Reply:
Good point. The comment started out somewhere else, so now `table` does indeed not make much sense anymore. It is the (height)`map` table that gets filled in the ReadHeightmap* functions.